Katello/katello
# Fixes #26600 - Duplicate host record when hypervisor_id changes

The intention of this change is to ensure we don't create orphaned host records in our database when a hypervisor that's checking in has modified its hypervisor_id in the virt-who config. Hypervisor records are already created based on the hypervisorId field, and this change will update the name on the host record if the hypervisorId has changed.

Candlepin has been updated so that it will key off of the dmi.system.uuid fact in the case that `hypervisorId` has changed and update *that* record instead of creating a new one. Since we are doing our side of the work after Candlepin we don't need to duplicate that lookup and work around it.

**Open questions:** 

- The downstream RFE asks that the host name is updated and that's what I've done here. Is that always desireable? Maybe we could have a setting that controls whether the host name will be updated. Should it be disabled or enabled by default?

**To test:**
- Update to Candlepin 2.6 locally (not strictly required, but it's the latest in nightly repos)
- Apply this change
- Using the fixture provided in this PR for the tests, issue a request to perform a hypervisor checkin:
```
curl -u "admin:changeme" -k -X POST -H 'Content-type: application/json' https://localhost/rhsm/hypervisors/Default_Organization -d '@./virt_who_altered_hypervisor_id.json'
```
- There should be two hosts in the UI
- Alter the hypervisorId field of the first hypervisor record in the fixture (notice that it has a dmi.system.uuid fact) to be anything else.
- Issue the hypervisor checkin request again.
- Check the UI and see there are still two hosts, but one of them has a new name reflecting the hypervisorId you modified
